client/mysqldump.c

```diff
@@ -6637,14 +6616,7 @@ static my_bool get_view_structure(char *table, char* db)
                 fix_for_comment(result_table));
 
   /* Table might not exist if this view was dumped with --tab. */
-  fprintf(sql_file, "/*!50001 DROP TABLE IF EXISTS %s*/;\n", opt_quoted_table);
-  if (opt_drop)
-  {
-    fprintf(sql_file, "/*!50001 DROP VIEW IF EXISTS %s*/;\n",
-            opt_quoted_table);
-    check_io(sql_file);
-  }
-
+  fprintf(sql_file, "/*!50001 DROP VIEW IF EXISTS %s*/;\n", opt_quoted_table);
```

Comment:
comment fixed. The `opt_drop` guard was dropped as this would be duplicate statements rather pointlessly. We are dropping out `select 1 as...` view and replacing it with our real one so this isn't optional.